plugins/catalog-backend/src/ingestion/LocationAnalyzer.ts

```diff
@@ -24,25 +26,51 @@ import {
 
 export class RepoLocationAnalyzer implements LocationAnalyzer {
   private readonly logger: Logger;
+  private readonly scmIntegrations: ScmIntegrations;
 
-  constructor(logger: Logger) {
+  constructor(logger: Logger, scmIntegrations: ScmIntegrations) {
     this.logger = logger;
+    this.scmIntegrations = scmIntegrations;
   }
   async analyzeLocation(
     request: AnalyzeLocationRequest,
   ): Promise<AnalyzeLocationResponse> {
-    const { owner, name, source } = parseGitUrl(request.location.target);
-    const entity = {
+    const { owner, name } = parseGitUrl(request.location.target);
+    const entity: Entity = {
       apiVersion: 'backstage.io/v1alpha1',
       kind: 'Component',
       metadata: {
         name: name,
-        // Probably won't handle properly self-hosted git providers with custom url
-        annotations: { [`${source}/project-slug`]: `${owner}/${name}` },
       },
       spec: { type: 'other', lifecycle: 'unknown' },
     };
 
+    const integration = this.scmIntegrations.byUrl(request.location.target);
+
+    let annotationPrefix;
+    switch (integration?.type) {
```

Comment:
I feel like resolving integration types to annotation prefixes, could most likely be something common. Would adding this to the `integration` package make sense?

Reply:
Good idea!